src/main/java/net/jung/chess/Board.java

```diff
@@ -1,85 +1,114 @@
+            blackPieceList.add(piece);
+    }
+
+    public void addPiece(Piece piece){
+        if(piece.getColor()==Color.WHITE) {
+            whitePieceList.add(piece);
+            return;
+        }
+        blackPieceList.add(piece);
     }
 
     public int size() {
-        return whitePawnList.size() + blackPawnList.size();
+        return whitePieceList.size() + blackPieceList.size();
     }
 
-    public int whitePawnListSize() {
-        return whitePawnList.size();
+    public int whitePieceListSize() {
+        return whitePieceList.size();
     }
 
-    public int blackPawnListSize() {
-        return blackPawnList.size();
+    public int blackPieceListSize() {
+        return blackPieceList.size();
     }
 
-    public Pawn findWhitePawn(int index) {
-        return whitePawnList.get(index);
+    public Piece findWhitePiece(int index) {
+        return whitePieceList.get(index);
     }
 
-    public Pawn findBlackPawn(int index) {
-        return blackPawnList.get(index);
+    public Piece findBlackPiece(int index) {
+        return blackPieceList.get(index);
     }
 
     public void initialize() {
-        int MAX_PAWNS_SIZE =8;
+        piecesExceptPawnInitialize(Color.BLACK);
+        pawnInitialize(Color.BLACK);
+
+        pawnInitialize(Color.WHITE);
+        piecesExceptPawnInitialize(Color.WHITE);
+    }
+
+    public void pawnInitialize(Color color){
+        int MAX_PAWNS_SIZE = 8;
         for (int i = 0; i < MAX_PAWNS_SIZE; i++) {
-            addWhitePawn(new Pawn(Color.WHITE));
-            addBlackPawn(new Pawn(Color.BLACK));
+            addPiece(Piece.createPiece(color, Name.PAWN));
         }
     }
 
-    public String getWhitePawnsResult() {
-        return getPawnsResult(whitePawnList);
+    public void piecesExceptPawnInitialize(Color color){
+        addPiece(Piece.createPiece(color, Name.ROOK));
+        addPiece(Piece.createPiece(color, Name.KNIGHT));
+        addPiece(Piece.createPiece(color, Name.BISHOP));
+        addPiece(Piece.createPiece(color, Name.QUEEN));
+        addPiece(Piece.createPiece(color, Name.KING));
+        addPiece(Piece.createPiece(color, Name.BISHOP));
+        addPiece(Piece.createPiece(color, Name.KNIGHT));
+        addPiece(Piece.createPiece(color, Name.ROOK));
+    }
+
+    public String getWhitePiecesResult() {
+        return getPiecesResult(whitePieceList);
     }
 
-    public String getBlackPawnsResult() {
-        return getPawnsResult(blackPawnList);
+    public String getBlackPiecesResult() {
+        return getPiecesResult(blackPieceList);
     }
 
-    public String getPawnsResult(List<Pawn> pawnList) {
+    public String getPiecesResult(List<Piece> pieceList) {
         StringBuilder pawnReps = new StringBuilder();
-        int colorPawnSize = pawnList.size();
-        for (int i = 0; i < colorPawnSize; i++) {
-            pawnReps.append(pawnList.get(i).getRepresentation());
+        for (int i = 0, colorPawnSize = pieceList.size(); i < colorPawnSize; i++) {
```

Comment:
```suggestion
        for (int i = 0; i < pieceList.size(); i++) {
```

또는 사실 이 size가 결국 8이기 때문에 `MAX_PAWNS_SIZE`를 `MAX_PIECES_SIZE` 정도로 바꾸고 활용하는 방안도 있겠네요.

Reply:
가독성은 조금 떨어지지만 for문 돌떄마다 size() 연산을 하는 것이 싫어 
for(int i = 0, colorPawnSize = pieceList.size(); i < colorPawnSize; i++)와 같이 작성했습니다.

size가 8이 아닐때도 고려해서 MAX_PIECES_SIZE를 쓰지 않고 size()를 넣었습니다!

Reply:
Size가 8이 아닌 경우는 어떤 경우인가요?
그렇게 되는 경우는 애플리케이션이 오류를 내야하는 것이 정상적인 동작 아닐까요?
차라리 연산을 하지 않도록 위로 꺼내주시는 것도 방법입니다.

Reply:
지난 단계에서 위로 꺼내는 방법을 썼었는데 간결하게 하라는 코멘트가 달려서 그에 따라 수정된 결과가 저것입니다ㅜㅜ 
size 8이 아닌 경우는 말이 죽었을 때가 있을 것 같은데 이번 미션에서는 그럴일은 없으니까 그냥 maxPiecesSize =16 으로  수정하는 것이 좋을것같네여 그렇게하겠습니다!